src/test/java/de/tum/in/www1/artemis/util/CourseTestService.java

```diff
@@ -1207,23 +1197,36 @@ public void testDownloadCourseArchiveAsInstructor_not_found() throws Exception {
 
     // Test
     public void testDownloadCourseArchiveAsInstructor() throws Exception {
+        submissionRepo.deleteAll();
 
-        // Dummy course archive
-        Path courseArchivePath = Path.of(courseArchivesDirPath, "some-course-archive.zip");
-        if (!Files.exists(courseArchivePath)) {
-            Files.createDirectories(Path.of(courseArchivesDirPath));
-            Files.createFile(courseArchivePath);
-        }
+        // Archive the course and wait until it's complete
+        testArchiveCourseWithTestModelingAndFileUploadExercises();
 
-        // Generate a course that has an archive
-        Course course = database.createCourse();
-        course.setCourseArchivePath(courseArchivePath.toString());
-        course = courseRepo.save(course);
+        var course = courseRepo.findAll().stream().findFirst().get();
+
+        // Download the archive
+        var archive = request.getFile("/api/courses/" + course.getId() + "/download-archive", HttpStatus.OK, new LinkedMultiValueMap<>());
+        assertThat(archive).isNotNull();
+        assertThat(archive).exists();
+
+        // Extract the archive
+        zipFileService.extractZipFileRecursively(archive.getAbsolutePath());
+        String extractedArchiveDir = archive.getPath().substring(0, archive.getPath().length() - 4);
+
+        // We test for the filenames of the submissions since it's the easiest way.
+        // We don't test the directory structure
+        var filenames = Files.walk(Path.of(extractedArchiveDir)).filter(Files::isRegularFile).map(Path::getFileName).collect(Collectors.toList());
+
+        var submissions = submissionRepo.findAll();
```

Comment:
Same here:
`/opt/bambooagent/bamboo-agent-home/xml-data/build-dir/ARTEMIS-TESTS1730-JAVATEST/src/test/java/de/tum/in/www1/artemis/util/CourseTestService.java:1220: error: cannot find symbol`